assets/src/date/test/getDateObjectWithTimezone.js

```diff
@@ -0,0 +1,342 @@
+
+describe('date/getDateObjectWithTimezone July', () => {
+  const MOCK_UTC_DATE = moment.utc('2020-07-12 11:30');
+
+  beforeEach(() => {
+    MockDate.set(MOCK_UTC_DATE);
+  });
+
+  it('should return UTC by default which is default moment', () => {
+    expect(getDateObjectWithTimezone()).toBe('2020-07-12T11:30:00Z');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when timezone is set to America/Los_Angeles', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: 'America/Los_Angeles',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when gmtOffset is -7 (America/Los_Angeles)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T07:30:00-04:00 when timezone is set to America/New_York', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -4,
+      timezone: 'America/New_York',
+    });
+
+    expect(date).toBe('2020-07-12T07:30:00-04:00');
+  });
+
+  it('should return 2020-07-12T07:30:00-04:00 when gmtOffset is -4 (America/New_York)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -4,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T07:30:00-04:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when timezone is set to America/Tijuana', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: 'America/Tijuana',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when gmtOffset is -7 (America/Tijuana)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T08:30:00-03:00 when timezone is set to America/Sao_Paulo', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -3,
+      timezone: 'America/Sao_Paulo',
+    });
+
+    expect(date).toBe('2020-07-12T08:30:00-03:00');
+  });
+
+  it('should return 2020-07-12T08:30:00-03:00 when gmtOffset is -3 (America/Sao_Paulo)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -3,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T08:30:00-03:00');
+  });
+
+  it('should return 2020-07-12T15:30:00+04:00 when timezone is set to Asia/Dubai', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 4,
+      timezone: 'Asia/Dubai',
+    });
+
+    expect(date).toBe('2020-07-12T15:30:00+04:00');
+  });
+
+  it('should return 2020-07-12T15:30:00+04:00 when gmtOffset is + 4 (Asia/Dubai)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 4,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T15:30:00+04:00');
+  });
```

Comment:
Hmmm... I'm a little confused by these tests.  The `timezone` test on line 111 still passes in the `gmtOffset`, but the `gmtOffset` test does _not_ pass in the `timezone`... is `timezone` even being used since it looks like `gmtOffset` is required and it alone can determine the correct time?

Reply:
Great point to clarify! 
So, the timezone is checked first when getting the right format, if it's present then the gmtOffset is ignored, gmtOffset is used as a backup because wordpress should always be passing that to us regardless of timezone. If all else fails, we're just formatting the utc time. 

timezone defaults to an empty string, gmtOffset defaults to null. 

I, doing copy and paste and trying to keep numbers straight, did not delete the gmtOffset where timezone is getting used because it was a very helpful sanity check to me as i set things up!  Now that everything is working though I am going ahead and fixing that so it's less confusing.